common/慎防一人團隊.md

```diff
@@ -0,0 +1,54 @@
+
+吃完中餐，我與其他實習生，很隨性地聊到我的”程式炸彈“（code bomb)。我對於我這個夏天所完成的專案感到沾沾自喜，但他們聽完，全部嚇死了。
+
+「你說什麼？！如果你的mentor找到明顯的設計錯誤怎麼辦？你的mentor有足夠的時間做全部的review嘛？而在他找到的前提下，你又有足夠的時間能解掉所有的issue嘛？如果他不決定讓你check in你極大的code commit？」
+
+我整顆心有如沉入大海。難道我整個夏天都將浪費掉了嗎？我花了我最後一個禮拜在 Google 擔心這件事會如何發展。
+
+所幸，我的 mentor 樂於助人，且在我實習階段結束後，願意處理需要面對的 issue。我得以 commit 我的 code，且這項功能在我離開後幾個月內正式運作。但有很多部分都需要找機會處理，且我的整個專案恐因此終止。回頭來看這件事，很明顯的，如果我頻繁地 commit 我一小段一小段的 code，我的成果不會被單獨孤立這麼久，且我可以去除大部分的風險。而我的 mentor 會有更多時間 review 我的 code，以至於我有更多有價值的回饋來改進我的 code。最後，我很幸運能在我職涯早期，因為獨立專案學到了這門課，而只付出少少的代價。
+
+有很多情況，是你必須獨立開發專案。有時候你需要嘗試遠離溝通開銷(communication overhead)，以及 `（不確定翻譯）僅一人團隊負責的專案`。另外的情況是：團隊會內部分成好幾個一人團隊，來單獨處理小的 task，來讓整體協作更容易。部分組織會主張他們的升遷過程，需要工程師證明他們能獨當一個專案。這樣能激勵工程師對於升遷做最大的努力。有些工程師就是喜歡能自己獨立作業。
+
+一人團隊開發專案，本質上是沒有問題，但仍然有些額外的風險容易降低專案的成功。首先也是最重要的，開發過程中很難獲得回饋(feedback)，而你需要回饋來驗證你的開發是可運行的。除非幫你做 code review 的工程師，跟你在同個團隊，且與你開發同個專案，否則你很難從 code reivew 中取得好的回饋。如果你對於 `取得回饋`(feedback loop)這件事不以為意，你有可能會較晚拿到回饋直到你以為你的 code 很完美。 而如果你到了專案尾端，都尚未發現你開發方向是錯的，就容易前功盡棄。
+
+一人團隊開發專案還有其他風險。當你獨立作業的時候，開發上的低潮較易使人感到喪志(demoralizing)。你必須面對下述狀況：困在你無法逃脫的沙坑，磨掉單調的工作，以及許多能藐視你理解能力的bug，如果有人能陪你分擔這些痛苦，能量比較不會這麼快耗盡，也比較能承受。只要一件事情卡住，便能停止專案進度，延後專案期限。我曾經經歷過，也看過別的工程師經歷過。只要一人團隊裡，還能額外配一個成員，縱使有人進度卡住了，整個團隊依然能維持著動力繼續前進。
+
+同樣的，當你獨立作業的時候，開發上的高潮較不易使人感到有動力。與團隊夥伴一起慶祝達到開發上的成就，是鼓動團隊士氣最好的方法(boost morale)。如果你是一人作業，當你終於解開了令人沮喪的 bug，誰能跟你擊掌？此外，當你知道，你的團隊夥伴是需要依靠你的時候，是會增加你的責任感(sense of accountability)。希望幫助夥伴達成目標的渴望，是可以讓每一個成員沉浸在開發的動力上(dips in motivation)。
+
+即使你仍然是一人團隊在作業，也不用絕望。上述的風險是可以被克服的。Steve Woziniak 最初在他家（之後到 Steve Jobs 的車庫）開發了 Apple I 與 Apple II 電腦，軟體與硬體皆自己獨立開發。當他的作品，自己興趣使然的玩具，從 Homebrew Computer Club 到創造了個人電腦革命，你怎麼看這件事情？一個關鍵的原因：Wozniak 有 Jobs 能給他平衡點(counterbalance)與回饋來驗證他的想法。縱使 Woziniak 內向，且表面上只做自己的東西，但他並沒有閉門造車，而是被 Jobs 的遠見與野心所鞭策了。最後，他們兩位新創了Apple公司。
+
+有如 Woziniak，我們可以建構出必要的回饋管道(necessary feedback channel)，來增加專案的成功率。以下是其策略：
+
+* 對於回饋保持開放與接受的心態
+如果你對於你工作內容保持防禦心態，你便很難將回饋聽進去，之後人們就很難有意願再幫忙。取而代之，要保持樂觀的心態去學習。不要將回饋與批評視為個人攻擊，而是視為改進自我的機會。
+
+* 及早且頻繁地 commit code
+更動量大的 code 是很難 review 的，需要花很多時間才能獲得回饋，如果結果是你開發根本偏了方向，浪費時間，浪費精力。專注在迭代進度，並利用迭代式 commit 來徵求回饋。不要成為發出極大 code review 的人。
+
+* 從透徹批評中要求 code review
+給不同的工程師做嚴格地 code review，是有很大的差異。如果你急著將 code 送出去，你也許會想將 code 交給會粗略 review 的工程師看過且批准。但如果你想優化 code 的品質，或者你想確保你的 code 可運作，你應該找能在 code review 中給予你深思熟慮批評的工程師。開發早期從團隊夥伴中獲得嚴厲的回饋，總比後期從用戶得到回饋得知某某功能無法使用來的好。
+
+* ask to bounce ideas off your teammates
+獲得回饋，最直接的途徑是：直接要求它。如果你能佔用對方幾分鐘的時間，在白板上討論你的想法，那就直接詢問正躺在飲水機旁的沙發上的團隊夥伴。研究指出：對其他人解釋一個想法，是自我學習最好的方法之一。此外，你的解釋也許會揭露你理解上的漏洞。大部分的人都想要幫忙，且對於能用片刻時間去接觸一個不同且有趣的問題感到感激。也就是說，如果你想要保持你的回饋管道未來仍暢通的話，要對你夥伴的時間感到尊敬。事前討論的準備要做足。確保你可以清楚描述：你嘗試解決的問題，以及你已經嘗試了哪些方法。討論過後，reciprocate with an offer to be a sounding board for their ideas.
```

Comment:
reciprocate with an offer to be a sounding board for their ideas.
作為回饋，你也可以在需要的時候為它們的想法給出意見。

- sounding board  
a person or group of people that you use to test something such as a new idea or suggestion to see if they will accept it or if they think it will work